Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -211,6 +211,43 @@ initialize addLinter cdotLinter
 
 end CDotLinter
 
+/-!
+# The `dollarSyntax` linter
+
+The `dollarSyntax` linter flags uses of `<|` that are achieved by typing `$`.
+-/
+
+/-- The `dollarSyntax` linter flags uses of `<|` that are achieved by typing `$`. -/
+register_option linter.dollarSyntax : Bool := {
+  defValue := true
```

Comment:
Note to self: when #15616 lands, make sure to also tweak this default. (At the moment, it's not clear which PR will land first.)